feat: add rocks for Slurm services (#1)
canonical/slurm-rocks: README.md

# Slurm rocks

The `slurm-rocks` repository is a collection of files to build rocks for the upstream
Slurm services.

## Getting started

> To get started with the project, you must have [Rockcraft] and [Just] installed for

Reviewer: [nit]: We could say "To get started with the Slurm rocks" here.

```suggestion
To get started with the Slurm rocks, you must have [Rockcraft] and [Just] installed for
```

Author: Fixed

packing the rocks, [Skopeo] for copying the generated OCI images to your local Docker registry, and
finally [Docker] for running the images.

```bash

> just pack # Packs all the rocks.
> just import # Imports all the rocks into your local Docker registry.
> docker run --rm -d --name slurmctld slurmctld:latest # Runs the slurmctld rock with Docker.

Reviewer: [nit]: Typically for inline comments I add two spaces instead of one. It's a Pythonism, but I find it's helpful for visually separating the comment from the command:

```suggestion
just pack  # Packs all the rocks.
just import  # Imports all the rocks into your local Docker registry.
docker run --rm -d --name slurmctld slurmctld:latest  # Runs the slurmctld rock with Docker.
```

Author: Fixed

```

A more extensive usage example can be seen on the (example)[./example] directory, which sets up all
the Slurm services into a simple cluster.

> [Rockcraft]: https://documentation.ubuntu.com/rockcraft
> [Just]: https://github.com/casey/just
> [Skopeo]: https://github.com/containers/skopeo
> [Docker]: https://www.docker.com

Reviewer: [nit]: Going off of what we have previously done, typically in our READMEs we refer to `just` in the lowercase. Also, look at the docs for `skopeo` on GitHub, it looks like it's preferred to refer to `skopeo` in lowercase as well. I think Rockcraft and Docker are fine capitalized since that's how their referred to in their respective docsets.

```suggestion
[Rockcraft]: https://documentation.ubuntu.com/rockcraft
[just]: https://github.com/casey/just
[skopeo]: https://github.com/containers/skopeo
[Docker]: https://www.docker.com
```

Author: Fixed
